qp_spades/qp_spades.py

```diff
@@ -6,12 +6,134 @@
+        if len(fps) != 2:
+            mgs = f'There are multiple files that match "{prefix}"'
+            return False, None, mgs
+        files.append('\t'.join([fps[0], fps[1], prefix]))
+
+    # 2. format main comand
+    # cmd = (f'{spades_cmd} > {out_dir}/%s.spades.log 2>&1 && '
+    #            f'mv {out_dir}/{sample_name}/scaffolds.fasta {out_fp}' % (
+    #             fwd_fp, rev_fp, sample_name, sample_name))
+    command = (
+        f'spades.py --{params["type"]} -t {ppn} -m {memory} '
+        f'-k {params["k-mers"]} -o $OUTDIR/$SNAME')
+    if params['merging'].startswith('flash '):
+        # get read lenght quickly; note that we are going to assume
+        # that (1) the forward and reverse are the same lenght and (2)
+        # all file pairs have the same lenght so only calculate once
+        fp = glob(join(directory, list(prefix_to_name)[0] + '*'))[0]
+        std_out, std_err, return_value = system_call(
+                f'gunzip -c {fp} | head -n 2')
+        if return_value != 0:
+            error_msg = (f"Error uncompressing: {fp}\n"
+                         f"Std out: {std_out}\nStd err: {std_err}\n")
+            return False, None, error_msg
+        read_length = len(std_out.split('\n')[1])
+        percentage = int(params['merging'][6:-1])/100
+        overlap = int(read_length * percentage)
+
+        command = (
+            # flash
+            f'flash --threads {ppn} --max-overlap={overlap} '
+            '--output-directory $OUTDIR '
+            '--output-prefix="$SNAME" ${FWD} ${REV} '
+            '--max-mismatch-density=0.1 > $OUTDIR/${SNAME}.flash.log 2>&1'
+            ' && '
+            # spades
+            f'{command} '
+            '--merge $OUTDIR/${SNAME}.extendedFrags.fastq '
+            '-1 $OUTDIR/${SNAME}.notCombined_1.fastq '
+            '-2 $OUTDIR/${SNAME}.notCombined_2.fastq')
+    else:
+        command = '%s -1 ${FWD} -2 ${REV}' % command
+
+    # 3. create qsub for array submission
+    num_samples = len(prefix_to_name)
+    mqsub = [
+        '#!/bin/bash',
+        '#PBS -M qiita.help@gmail.com',
+        f'#PBS -N {job_id}',
+        f'#PBS -l nodes=1:ppn={ppn}',
+        f'#PBS -l walltime={WALLTIME}',
+        f'#PBS -l mem={memory}g',
+        f'#PBS -o {output_dir}/{job_id}' + '_${PBS_ARRAYID}.log',
+        f'#PBS -e {output_dir}/{job_id}' + '_${PBS_ARRAYID}.err',
+        f'#PBS -t 1-{num_samples}%{MAX_RUNNING}',
+        '#PBS -l epilogue=/home/qiita/qiita-epilogue.sh',
+        f'cd {output_dir}',
+        f'{environment}',
+        f'OUTDIR={output_dir}/',
+        'date',
+        'hostname',
+        'echo ${PBS_JOBID} ${PBS_ARRAYID}',
+        'offset=${PBS_ARRAYID}',
+        'args=$(head -n $offset ${OUTDIR}/files_to_process.txt| tail -n 1)',
+        "FWD=$(echo -e $args | awk '{ print $1 }')",
+        "REV=$(echo -e $args | awk '{ print $2 }')",
+        "SNAME=$(echo -e $args | awk '{ print $3 }')",
+        f'{command}',
+        'date']
```

Comment:
In addition to wrapping in `date`, I wonder if it may be pleasant to also use `f'/usr/bin/time -v {command}`? It will capture more detail about the runtime, peak memory and other pieces

Reply:
Excellent question! I think we are already capturing this information via the `epilogue=/home/qiita/qiita-epilogue.sh`; however, if you think is worth adding, I'll add!

Reply:
thanks!